test/suites/waitready.sh

```diff
@@ -0,0 +1,83 @@
+test_waitready() {
+  local LXD_STORAGE_DIR
+
+  # Spawn a new daemon.
+  LXD_STORAGE_DIR=$(mktemp -d -p "${TEST_DIR}" XXXXXXXXX)
+  spawn_lxd "${LXD_STORAGE_DIR}" false
+  LXD_DIR="${LXD_STORAGE_DIR}"
+
+  # Create storage pool.
+  local lxd_backend storage_pool
+  lxd_backend=$(storage_backend "$LXD_DIR")
+  storage_pool="lxdtest-$(basename "${LXD_DIR}")-pool"
+  br_name="lxdt$$"
+
+  lxc storage create "${storage_pool}" "${lxd_backend}"
+  lxc network create "${br_name}"
+
+  # Corrupt the storage pool directory to cause errors when LXD starts trying to start the storage pools.
+  # Replace the directory with a simple file which should cause an error.
+  rm -r "${LXD_DIR}/storage-pools/${storage_pool}"
+  touch "${LXD_DIR}/storage-pools/${storage_pool}"
+
+  # Restart the daemon.
+  shutdown_lxd "${LXD_DIR}"
+  respawn_lxd "${LXD_DIR}" true
+
+  # Wait 10s for storage.
+  # LXD started but fails to start all storage pools.
+  ! lxd waitready --storage --timeout 10 | grep -q "Error: Storage pools not ready yet after 10s timeout"
+
+  # Not setting a timeout should have the same effect and return instantly.
+  ! lxc query "/internal/ready?storage=1" | grep -q "Error: Storage pools not ready yet"
+
+  # The standard waitready without extra flags should still return with success.
+  lxd waitready
+
+  # Restore the storage pool directory.
+  rm "${LXD_DIR}/storage-pools/${storage_pool}"
+  mkdir "${LXD_DIR}/storage-pools/${storage_pool}"
+
+  # Restart the daemon.
+  shutdown_lxd "${LXD_DIR}"
+  respawn_lxd "${LXD_DIR}" true
+
+  # Wait for storage.
+  # This time it should return with success as LXD is able to start the storage pool.
+  lxd waitready --storage --timeout 10
+
+  # Corrupt the network by setting an invalid MTU.
+  lxd sql global 'INSERT INTO networks_config (network_id, node_id, key, value) VALUES (1, null, "bridge.mtu", "42")'
+
+  # Restart the daemon.
+  shutdown_lxd "${LXD_DIR}"
+  respawn_lxd "${LXD_DIR}" true
+
+  # Wait 10s for network.
+  # LXD started but fails to start all networks.
+  ! lxd waitready --network --timeout 10 | grep -q "Error: Networks not ready yet after 10s timeout"
```

Comment:
Could we get away with shorter timeouts when failures are expected? I'm asking because the average runtime of our test functions is 13s (but it's heavily skewed by `shutdown` taking 5.5 minutes :/).